### PR TITLE
Update version of azure-extension-platform | Add Telemetry Events to Extension 

### PR DESCRIPTION
**Pull request by simratsatia**
Why is this PR required?
- Fetch latest version of azure-extension-platform that contains critical bug fix for extension install.
Install command returning success even in case of failure
Bug: https://github.com/Azure/azure-extension-platform/commit/67e58652b30d119aa84715cb48e290b6fd714543
Bug: Use correct version of extension during logging: https://github.com/Azure/azure-extension-platform/commit/0dc957a76b678ea385889b0f1b123d5be58994b8
- Add extension events for better telemetry

Changes made:
- Using ext.ExtensionEvents.LogInformationalEvent/LogErrorEvents, we are sending the telemetry events
- Updated go.mod/go.sum with latest version of azure-extension-platform by recreating these files

### CODE REVIEW

**Review by rjschwei — APPROVED** [review]

**Comment by rjschwei** [comment]
@jesusbv please take a look as well.

File: main/ahb_for_sles.go [hunk 1]
```diff
@@ -421,13 +468,17 @@ func getExtensionAndRun() error {
 	initilizationInfo.UpdateCallback = updateCallbackFunc
 	vmExt, err := getVMExtensionFuncToCall(initilizationInfo)
 	if err != nil {
+		fmt.Fprintln(os.Stderr, "Extension initialization failed. Reason="+err.Error())
+		vmExt.ExtensionEvents.LogErrorEvent(
+			"Initialization",
+			fmt.Sprintf("AHBForSLES extension initialization finished. Result=Failure; Reason=%v", err.Error()))
```
Comment (jesusbv) [inline main/ahb_for_sles.go:474]: shouldn't we use `OPERATION_FAILURE_MSG` here ?
```go
fmt.Sprintf(OPERATION_FAILURE_MSG, "Initialization", err.Error())
```

File: main/ahb_for_sles.go [hunk 2]
```diff
@@ -421,13 +468,17 @@ func getExtensionAndRun() error {
 	initilizationInfo.UpdateCallback = updateCallbackFunc
 	vmExt, err := getVMExtensionFuncToCall(initilizationInfo)
 	if err != nil {
+		fmt.Fprintln(os.Stderr, "Extension initialization failed. Reason="+err.Error())
+		vmExt.ExtensionEvents.LogErrorEvent(
+			"Initialization",
```
Comment (jesusbv) [inline main/ahb_for_sles.go:473]: I understand `Initialization` is not an operation available/listed in the HandlerManifest but maybe we could add 
`INITIALIZE_EVENT:     "Initialization"` to the [consts in line 49](https://github.com/SUSE-Enceladus/ahb-extension/pull/15/files#diff-4fd45add4912cb8303f57a306b692c86cdb8e0ec31ad3d564b867f725eb84161R49) ?

**Review by jesusbv — COMMENTED** [review]
@simratsatia thanks for the PR to improve the telemetry

I made a couple of small suggestions

File: main/ahb_for_sles.go (hunk 1 above)
Comment (simratsatia) [reply main/ahb_for_sles.go:474]: Fair point @jesusbv

File: main/ahb_for_sles.go (hunk 2 above)
Comment (simratsatia) [reply main/ahb_for_sles.go:473]: Pushed the change

**Review by jesusbv — APPROVED** [review]

File: main/ahb_for_sles.go (hunk 2 above)
Comment (jesusbv) [reply main/ahb_for_sles.go]: Thank you for the changes

**Pull request merged**